### DISCUSSION TITLE
MQTT Topics

### DISCUSSION THREAD

**Discussion by psfshr** [Koenkk/zigbee2mqtt#7914] [General]
Is there any way to separate device messages into a separate MQTT sub-topic?

e.g. at the moment if I have devices named Sensor1 and Sensor2 I see the following MQTT topics published:

zigbee2mqtt/bridge
zigbee2mqtt/Sensor1
zigbee2mqtt/Sensor2

What I'd like to do is to put the sensors under a separate sub-topic, i.e.

zigbee2mqtt/bridge
zigbee2mqtt/devices/Sensor1
zigbee2mqtt/devices/Sensor2

The reasoning for this is that I'm using Telegraf to push the device data into InfluxDB for graphing. Currently I have Telegraf set to subscribe to "zigbee2mqtt/#" so that I don't need to subscribe to every individual device, but it means that I also get all the bridge messages in InfluxDB as well. If it was possible to have devices under a sub-topic, then I could set Telegraf to subscribe to "zigbee2mqtt/devices/#" and just record the sensor data.

**Comment by sjorge** [comment]
You can use / in the friendly name just fine.

So you would name your device `devices/Sensor1` instead of `Sensor1`

**Reply by psfshr** [reply]
Ok. I'll give that a try. Thanks!

**Reply by psfshr** [reply]
Works.

**Comment by satter** [comment]
I've set up Telegraf initially with
```
   [inputs.mqtt_consumer.tagdrop]
     topic = ["zigbee2mqtt/bridge/*"]
```
Telegraf still tries to read all available topics generating bunch of errors in logs but it filters out all bridge related messages
